barretenberg/cpp/src/barretenberg/vm/avm/trace/execution.cpp

```diff
@@ -281,23 +268,31 @@ std::vector<Row> Execution::gen_trace(std::vector<Instruction> const& instructio
  * @param public_inputs expressed as a vector of finite field elements.
  * @return The trace as a vector of Row.
  */
-std::vector<Row> Execution::gen_trace(std::vector<Instruction> const& instructions,
-                                      std::vector<FF>& returndata,
+std::vector<Row> Execution::gen_trace(std::vector<uint8_t> const& bytecode,
                                       std::vector<FF> const& calldata,
                                       std::vector<FF> const& public_inputs_vec,
+                                      std::vector<FF>& returndata,
                                       ExecutionHints const& execution_hints)
 
 {
+    std::vector<Instruction> instructions = Deserialization::parse(bytecode);
+    vinfo("Deserialized " + std::to_string(instructions.size()) + " instructions");
     vinfo("------- GENERATING TRACE -------");
     // TODO(https://github.com/AztecProtocol/aztec-packages/issues/6718): construction of the public input columns
     // should be done in the kernel - this is stubbed and underconstrained
-    VmPublicInputs public_inputs = convert_public_inputs(public_inputs_vec);
+    VmPublicInputs public_inputs = avm_trace::convert_public_inputs(public_inputs_vec);
     uint32_t start_side_effect_counter =
         !public_inputs_vec.empty() ? static_cast<uint32_t>(public_inputs_vec[START_SIDE_EFFECT_COUNTER_PCPI_OFFSET])
                                    : 0;
-
-    AvmTraceBuilder trace_builder =
-        Execution::trace_builder_constructor(public_inputs, execution_hints, start_side_effect_counter, calldata);
+    std::vector<std::vector<uint8_t>> all_contract_bytecode;
+    all_contract_bytecode.reserve(execution_hints.externalcall_hints.size() + 1);
```

Comment:
Is `vector.reserve()` more efficient than alternatives?

Reply:
Just curious!

Reply:
i believe it just doesn't do the allocation jsut updates the `capacity` of the vector as opposed to `resize` which would allocate some default value

Reply:
Ah nice!